### PR TITLE
Authorization Code Flow for Single Page Applications: SSO Fixes and changes

### PR DESCRIPTION
**Pull request by pkanher617**
This PR adds ADAL SSO and fixes other SSO scenarios for the Auth Code library.

Please review PR #1208 before this PR for better context.

### CODE REVIEW

File: lib/msal-common/src/app/module/AuthorizationCodeModule.ts [hunk 1]
```diff
@@ -84,15 +84,20 @@ export class AuthorizationCodeModule extends AuthModule {
             );
 
             // Check for SSO
+            let adalIdToken: IdToken;
```
Comment (jmckennon) [inline lib/msal-common/src/app/module/AuthorizationCodeModule.ts]: should we really be naming things with `adal`?

File: samples/VanillaJSTestApp/index_authCode.html [hunk 2]
```diff
@@ -55,7 +55,7 @@
     // signin and acquire a token silently with POPUP flow. Fall back in case of failure with silent acquisition to popup
     function signIn() {
         try {
-            myMSALObj.loginRedirect(loginRequest);
+            myMSALObj.acquireTokenRedirect(loginRequest);
```
Comment (jasonnutter) [inline samples/VanillaJSTestApp/index_authCode.html]: Mean to commit this?

File: lib/msal-common/src/app/module/AuthorizationCodeModule.ts (hunk 1 above)
Comment (jasonnutter) [reply lib/msal-common/src/app/module/AuthorizationCodeModule.ts]: This is for SSOing to an ADAL session, so yes, in this case.

File: samples/VanillaJSTestApp/index_authCode.html (hunk 2 above)
Comment (pkanher617) [reply samples/VanillaJSTestApp/index_authCode.html]: No, will update this in pre-alpha PR.

File: lib/msal-common/src/app/module/AuthorizationCodeModule.ts (hunk 1 above)
Comment (pkanher617) [reply lib/msal-common/src/app/module/AuthorizationCodeModule.ts]: As Jason said, this is a one-off specifically for ADAL, so doesn't get used otherwise.

**Pull request merged**